packages/typespec-ts/test/util/emitUtil.ts

```diff
@@ -562,3 +567,21 @@ export async function emitModularClientFromTypeSpec(
   expectDiagnosticEmpty(dpgContext.program.diagnostics);
   return undefined;
 }
+
+export async function emitSamplesFromTypeSpec(
+  tspContent: string,
+  examples: Record<string, string>,
+  configs: Record<string, string> = {}
+) {
+  const context = await compileTypeSpecFor(tspContent, examples);
+  const dpgContext = await createDpgContextTestHelper(context.program, false, {
+    "examples-directory": `./examples`,
```

Comment:
not sure why we need this in ut ?

Reply:
this is not ut but helper to help build samples and is used when we do sample scenarios testings.